src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,174 @@
+    }
+
+    const url = !isDisabled ? createURL(pageNumber) : '#';
+
+    return (
+      <PaginationLink
+        ariaLabel={ariaLabel}
+        cssClasses={cssClasses}
+        handleClick={this.handleClick}
+        isDisabled={isDisabled}
+        key={label + pageNumber + ariaLabel}
+        label={label}
+        pageNumber={pageNumber}
+        url={url}
+      />
+    );
+  }
+
+  public handleClick = (pageNumber: number, event: MouseEvent) => {
+    if (isSpecialClick(event)) {
+      // do not alter the default browser behavior
+      // if one special key is down
+      return;
+    }
+    event.preventDefault();
+    this.props.setCurrentPage(pageNumber);
+  };
+
+  private previousPageLink = () => {
+    return this.pageLink({
+      ariaLabel: 'Previous',
+      additionalClassName: this.props.cssClasses.previousPageItem,
+      isDisabled: this.props.isFirstPage,
+      label: this.props.templates.previous,
+      pageNumber: this.props.currentPage - 1,
+      createURL: this.props.createURL,
+    });
+  };
+
+  private nextPageLink = () => {
+    return this.pageLink({
+      ariaLabel: 'Next',
+      additionalClassName: this.props.cssClasses.nextPageItem,
+      isDisabled: this.props.isLastPage,
+      label: this.props.templates.next,
+      pageNumber: this.props.currentPage + 1,
+      createURL: this.props.createURL,
+    });
+  };
+
+  private firstPageLink = () => {
+    return this.pageLink({
+      ariaLabel: 'First',
+      additionalClassName: this.props.cssClasses.firstPageItem,
+      isDisabled: this.props.isFirstPage,
+      label: this.props.templates.first,
+      pageNumber: 0,
+      createURL: this.props.createURL,
+    });
+  };
+
+  private lastPageLink = () => {
+    return this.pageLink({
+      ariaLabel: 'Last',
+      additionalClassName: this.props.cssClasses.lastPageItem,
+      isDisabled: this.props.isLastPage,
+      label: this.props.templates.last,
+      pageNumber: this.props.nbPages! - 1,
+      createURL: this.props.createURL,
+    });
+  };
+
+  private pages = () => {
+    return this.props.pages!.map(pageNumber =>
+      this.pageLink({
+        ariaLabel: `${pageNumber + 1}`,
+        additionalClassName: this.props.cssClasses.pageItem,
+        isSelected: pageNumber === this.props.currentPage,
+        label: `${pageNumber + 1}`,
+        pageNumber,
+        createURL: this.props.createURL,
+      })
+    );
+  };
+
+  public render() {
+    return (
+      <div
+        className={cx(this.props.cssClasses.root, {
+          [this.props.cssClasses.noRefinementRoot]: this.props.nbPages! <= 1,
+        })}
+      >
+        <ul className={this.props.cssClasses.list}>
+          {this.props.showFirst && this.firstPageLink()}
+          {this.props.showPrevious && this.previousPageLink()}
+          {this.pages()}
+          {this.props.showNext && this.nextPageLink()}
+          {this.props.showLast && this.lastPageLink()}
+        </ul>
+      </div>
+    );
+  }
+}
+
```

Comment:
defaultProps have been removed, does that make a difference in usage anywhere (a prop that isn't given or something?)

Reply:
it's defined here: https://github.com/algolia/instantsearch.js/pull/4765/files/5fa4d4c3ead6aeda1745bdb83792262539164fab#diff-d28dbf12c34fc4ebf83f4564d72917e83e74efd40d6056d2889fb087efc80537R43-R50

We also use this syntax in other components such as `SearchBox`, do you think we should use `ComponentName.defaultProps`?

Reply:
I missed that bit, perfectly logical to me!